tutorials/Flop_Count_Example.ipynb

```diff
@@ -0,0 +1,327 @@
+    "3. The third item returned by the counter records all the operations that the counter didn't know how to count flops for (more detail about these ops below). This dicitionary (or more specifically a [collections.Counter](https://docs.python.org/3/library/collections.html#collections.Counter)) maps unknown op names to the number of times those ops were called in the model. In our example, all of the model's operations are supported by the counter, so this dictionary is empty.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "2be3fb9f",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(19619840,\n",
+       " {'conv': 18595840, 'batch_norm': 614400, 'linear': 409600},\n",
+       " Counter())"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "170ea68b",
+   "metadata": {},
+   "source": [
+    "In general, the majority of common ML operations are already supported, although some operations' counts may be approximations (e.g. one floating point addition and one square root are both counted as 1 flop). Much more complicated models than the toy example above can be evaluated without accumulating lots of unsupported operations, such as the [scale hyperprior model](https://arxiv.org/abs/1802.01436):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "961a052c",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "bigger_model = ScaleHyperprior(network_channels=32, compression_channels=64)\n",
+    "_,_, unsupported_ops = ncF.count_flops(bigger_model, (torch.randn(1, 3, 64, 64),))\n",
+    "\n",
+    "len(unsupported_ops) == 0 # Verifying no unsupported operations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3ce26204",
+   "metadata": {},
+   "source": [
+    "However, if you need to add support for more ops or override the counter's default implementation, read on!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52f36115",
+   "metadata": {},
+   "source": [
+    "## Advanced Usage\n",
+    "\n",
+    "### How the Counter Works\n",
+    "\n",
+    "The flop counter in `neuralcompression` makes heavy use of the counting utilities in [fvcore](https://github.com/facebookresearch/fvcore). Counting a model's flops is a two-step process:\n",
+    "\n",
+    "1. Using PyTorch's [TorchScript](https://pytorch.org/docs/stable/jit.html) capabilities, the model is first JIT-traced into a computational graph. Each node in the graph corresponds to an ATen (linear algebra) operation, like matrix multiplications, convolutions, and elementwise operations like additions/subtractions.\n",
```

Comment:
Won't the trace be correct given the specific inputs passed to the model? As in, the model's control flow logic won't actually be part of the traced graph, but the specific code paths followed for the provided inputs will be. In the context of flop counting, isn't this ok (since we're counting model flops with respect to a specific set of inputs / code path)?

Reply:
Hmm I guess mentioning what I said will add to the confusion. Let's let it be :)